src/autocomplete/typeahead.js

```diff
@@ -417,7 +418,11 @@ _.mixin(Typeahead.prototype, {
     if (typeof datum.value !== 'undefined') {
       this.input.setQuery(datum.value);
     }
-    this.input.setInputValue(datum.value, true);
+    if (this.clearOnSelected) {
+      this.input.setInputValue('', true);
```

Comment:
Add setQuery here too

Reply:
add `this.input.setQuery('');` here and we gucci :+1: